lib/PhasarLLVM/ControlFlow/LLVMBasedBackwardICFG.cpp

```diff
@@ -105,6 +104,52 @@ LLVMBasedBackwardsICFG::allNonCallStartNodes() const {
   return ForwardICFG.allNonCallStartNodes();
 }
 
+const llvm::Function *
+LLVMBasedBackwardsICFG::getFunctionOf(const llvm::Instruction *Stmt) const {
+  auto BackwardRetIt = BackwardRetToFunction.find(Stmt);
+  if (BackwardRetIt != BackwardRetToFunction.end()) {
+    return BackwardRetIt->second;
+  }
+  return Stmt->getFunction();
+}
+
+std::vector<const llvm::Instruction *>
+LLVMBasedBackwardsICFG::getPredsOf(const llvm::Instruction *Stmt) const {
+  auto BackwardRetIt = BackwardRetToFunction.find(Stmt);
+  if (BackwardRetIt == BackwardRetToFunction.end()) {
+    return LLVMBasedBackwardCFG::getPredsOf(Stmt);
+  }
+  auto ExitPoints =
+      LLVMBasedBackwardCFG::getExitPointsOf(BackwardRetIt->second);
+  std::vector<const llvm::Instruction *> Result(ExitPoints.begin(),
+                                                ExitPoints.end());
+  return Result;
+}
+
+std::vector<const llvm::Instruction *>
+LLVMBasedBackwardsICFG::getSuccsOf(const llvm::Instruction *Stmt) const {
+  if (Stmt->getParent() == nullptr && BackwardRetToFunction.count(Stmt) > 0) {
+    return {};
+  }
+  std::vector<const llvm::Instruction *> Succs =
+      LLVMBasedBackwardCFG::getSuccsOf(Stmt);
+  if (Succs.size() == 0) {
+    Succs.push_back(
+        BackwardRets.at(Stmt->getParent()->getParent()).getInstance());
```

Comment:
I don't think that 'just throwing' an exception is a good solution for cases where a error is foreseeable and should actually be handled on the design side. Especially, because LLVM by default tries to build without exceptions, hence LLVM code is not aware of exceptions at any point and does not follow exception usage pattern.

However, if you think that this case never happens because all functions are correctly registered in the IRDB, keep it like this.